llama_index/objects/base.py

```diff
@@ -64,3 +68,48 @@ def as_retriever(self, **kwargs: Any) -> ObjectRetriever:
 
     def as_node_retriever(self, **kwargs: Any) -> BaseRetriever:
         return self._index.as_retriever(**kwargs)
+
+    def persist(
+        self,
+        persist_dir: str = DEFAULT_PERSIST_DIR,
+        obj_node_mapping_fname: str = DEFAULT_PERSIST_FNAME,
+    ) -> None:
+        # try to persist object node mapping
+        try:
+            self._object_node_mapping.persist(
+                persist_dir=persist_dir, obj_node_mapping_fname=obj_node_mapping_fname
+            )
+        except (NotImplementedError, pickle.PickleError) as err:
+            warnings.warn(
+                (
+                    "Unable to persist ObjectNodeMapping. You will need to "
+                    "reconstruct the same object node mapping to build this ObjectIndex"
+                ),
+                stacklevel=2,
+            )
+        self._index._storage_context.persist(persist_dir=persist_dir)
+
+    @classmethod
+    def from_persist_dir(
+        cls,
+        persist_dir: str = DEFAULT_PERSIST_DIR,
+        object_node_mapping: Optional[BaseObjectNodeMapping] = None,
+    ) -> "ObjectIndex":
+        from llama_index.indices import load_index_from_storage
+
+        storage_context = StorageContext.from_defaults(persist_dir=persist_dir)
+        index = load_index_from_storage(storage_context)
+        if object_node_mapping:
+            return cls(index=index, object_node_mapping=object_node_mapping)
+        else:
+            # try to load object_node_mapping
+            try:
+                object_node_mapping = BaseObjectNodeMapping.from_persist_dir(
```

Comment:
Should it assume `SimpleObjectNodeMapping` here, since that's the only node mapping class we have?

Reply:
yup, let's do that!